src/wallet/wallet.cpp

```diff
@@ -2648,7 +2651,10 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
     if (GetRandInt(10) == 0)
         txNew.nLockTime = std::max(0, (int)txNew.nLockTime - GetRandInt(100));
 
-    assert(txNew.nLockTime <= (unsigned int)chainActive.Height());
+    {
+        LOCK(cs_main);
+        assert(txNew.nLockTime <= (unsigned int)chainActive.Height());
```

Comment:
I think we should just get rid of this assert.  We can't guarantee chainActive height will not go down.